digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/NodesAndJunctionsService.scala

```diff
@@ -158,11 +158,41 @@ class NodesAndJunctionsService(roadwayDAO: RoadwayDAO, roadwayPointDAO: RoadwayP
     }
   }
 
-  private def handleJunctionPointUpdate(junctionId: Long, oldJunctionPoint: JunctionPoint, newJunctionPoint: JunctionPoint, username: String): JunctionPoint = {
-    // TODO Check that the address change is within acceptable boundaries:
-    // - Less than 10 meters from the address calculated at this point if there were no calibration point here
-    // - Within the address range of the neighbouring links
+  // TODO remove this function and its usages when VIITE-2524 gets implemented
+  def areJunctionPointsOnRoadwayChangingSpot(junctionPointIds: Seq[Long]): Boolean = {
+    withDynSession {
+      val junctionPoints = junctionPointDAO.fetchByIds(junctionPointIds)
+      val roadwayPoints = junctionPoints.map(jp => roadwayPointDAO.fetch(jp.roadwayPointId))
+      val roadwayNumbers = roadwayPoints.map(rwp => rwp.roadwayNumber).toSet
+      val roadways = roadwayDAO.fetchAllByRoadwayNumbers(roadwayNumbers)
+      if (roadways.size < 2) {
+        false
+      } else {
+        true
+      }
+    }
+  }
 
+  def areJunctionPointsOnAdministrativeClassChangingSpot(junctionPointIds: Seq[Long]): Boolean = {
+    withDynSession {
+      val junctionPoints = junctionPointDAO.fetchByIds(junctionPointIds)
+      val roadwayPoints = junctionPoints.map(jp => roadwayPointDAO.fetch(jp.roadwayPointId))
+      val roadwayNumbers = roadwayPoints.map(rwp => rwp.roadwayNumber).toSet
```

Comment:
Tulee tässä näitä jp:iden perusteella roadwayita toistamiseen hakemista nähdess mieleen, olisiko meillä jo codebaessa, tai olisiko syytä, tehdä näistä hauista omat funkkarinsa.
Veikkaan etteivät nämä ole ainoat funktiot, joissa olisi tarpeen..?

Reply:
En ainakaan roadwayDAO:ssa nähnyt sellaista. Ja toinen näistä funtioista on tilapäinen niin tein sen mahdollisimman helposti poistettavaksi kun VIITE-2524 implementoidaan. Toistuvuus katoaa silloin myös.

Reply:
Mut meinaatko myös, että muualla Viiteessä ei ole ollut tollaisille tarvetta?
(Eli pelkän tämän tiketin kontekstissa ok just noin, just noista sun kertomistasi syistä johtuen. Mut mä veikkaan, ettei toi oo ensimmäinen kerta, kun tollasia ketjuhakuja on tarvittu, enkä usko että viimeinenkään. (Nimimerkillä "ittekki kironnu jp - j - node - np - rw -hakuketjuja")

Reply:
roadwaypoint haku taitaa olla turha koska junction pointilla on jo rwn mutta ei varmaan haittaa joskun funkkari on väliaikanen.

Reply:
Ahaa. No siinä tapauksessa tosta saa paljon simppelimmän!